src/test/testCognicityServer.js

```diff
@@ -0,0 +1,676 @@
+                }
+             });
+          });
+
+        // Catch invalid floodgauge id
+        it('Return 404 for get /floodgauges/:id', function(done){
+            test.httpAgent(app)
+              .get('/floodgauges/0')
+              .expect(404)
+              .expect('Content-Type', /json/)
+              .end(function(err, res){
+                if (err) {
+                  test.fail(err.message + ' ' + JSON.stringify(res));
+                }
+                else {
+                  done();
+                }
+             });
+          });
+    });
+
+    // Infrastructure endpoint
+    describe('Infrastructure endpoint', function() {
+      // Catch invalid top-level infrastructure endpoint
+      it('Return 404 for get /infrastructure', function(done){
+          test.httpAgent(app)
+            .get('/infrastructure')
+            .expect(404)
+            .expect('Content-Type', /json/)
+            .end(function(err, res){
+              if (err) {
+                test.fail(err.message + ' ' + JSON.stringify(res));
+              }
+              else {
+                done();
+              }
+           });
+        });
+
+        // Return waterways infrastructure
+        it('Return 200 for get /infrastructure/waterways', function(done){
+            test.httpAgent(app)
+              .get('/infrastructure/waterways')
+              .expect(200)
+              .expect('Content-Type', /json/)
+              .end(function(err, res){
+                if (err) {
+                  test.fail(err.message + ' ' + JSON.stringify(res));
+                }
+                else {
+                  done();
+                }
+             });
+          });
+    });
+
+    // Floods endpoint
+    describe('Flood areas endpoint', function(){
+
+      // Test put flood auth
+      it ('Catch bad auth for put a flood (PUT /floods/:id)', function(done){
+        test.httpAgent(app)
+          .put('/floods/5')
+          .send({
+              "state": "2"
+          })
+          .expect(401)
+          .expect('Content-Type', /json/)
+          .end(function(err, res){
+            if (err){
+              test.fail(err.message + ' ' + JSON.stringify(res));
+            }
+            else {
+              done();
+            }
+          });
+      });
+
+      // Test delete flood auth
+      it ('Catch bad auth for put a flood (PUT /floods/:id)', function(done){
+        test.httpAgent(app)
+          .delete('/floods/5')
+          .send({
+              "username": "testing"
+          })
+          .expect(401)
+          .expect('Content-Type', /json/)
+          .end(function(err, res){
+            if (err){
+              test.fail(err.message + ' ' + JSON.stringify(res));
+            }
+            else {
+              done();
+            }
+          });
+      });
+
+      // Get floods
+      it('Get floods (GET /floods)', function(done){
+        this.timeout(15000); // a lot of data is returned
```

Comment:
this is for mocha tests on travis which run slow because of small containers (the neighborhood polygons generate a ~2k rows of data).